rcp/org.testeditor.rcp4.uatests/src/org/testeditor/fixture/swt/SWTFixture.java

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2015 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.fixture.swt;
+
+import static org.eclipse.swtbot.swt.finder.matchers.WidgetMatcherFactory.widgetOfType;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Table;
+import org.eclipse.swt.widgets.Widget;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.testeditor.fixture.core.interaction.FixtureMethod;
+
+public class SWTFixture {
+
+	Logger logger = Logger.getLogger(SWTFixture.class);
+	SWTWorkbenchBot bot = new SWTWorkbenchBot();
+
+	/**
+	 * Method for debugging the AUT. A first implementation of an widget
+	 * scanner. all widget s on current SWT-App will be printed with their id's
+	 * in the swtbot.log with the trace-level. This Method is in work and can be
+	 * extended and modified step by step.
+	 * 
+	 */
+	@FixtureMethod
+	public void reportWidgets() {
+		logger.trace("analyzeWidgets start");
+		logger.trace("---------------------------------------------");
+
+		Display.getDefault().syncExec(new Runnable() {
```

Comment:
We should have a helper method for calling `Display.getDefault()` right from the beginning. This nesting in this method is quite deep which makes it hard to read.